blockdag/process.go

```diff
@@ -168,6 +168,22 @@ func (dag *BlockDAG) ProcessBlock(block *util.Block, flags BehaviorFlags) (isOrp
 		}
 	}
 
+	// Handle the case of a block with a valid timestamp(non-delayed) which points to a delayed block.
+	dag.delayedBlocksLock.RLock()
+	delay = 0
+	for _, parentHash := range block.MsgBlock().Header.ParentHashes {
+		if delayedParent, exists := dag.delayedBlocks[*parentHash]; exists {
+			parentDelay := delayedParent.processTime.Sub(dag.timeSource.AdjustedTime())
+			if parentDelay > delay {
+				delay = parentDelay
```

Comment:
Fix to `parentDleay + 1nanosecond` (pseudo code)

Reply:
Added below (line 185)